homeassistant/components/doorbird.py

```diff
@@ -34,11 +43,36 @@ def setup(hass, config):
     if status[0]:
         _LOGGER.info("Connected to DoorBird at %s as %s", device_ip, username)
         hass.data[DOMAIN] = device
-        return True
     elif status[1] == 401:
         _LOGGER.error("Authorization rejected by DoorBird at %s", device_ip)
         return False
     else:
         _LOGGER.error("Could not connect to DoorBird at %s: Error %s",
                       device_ip, str(status[1]))
         return False
+
+    if config[DOMAIN].get(CONF_DOORBELL_EVENTS):
+        # Provide an endpoint for the device to call to trigger events
+        hass.http.register_view(DoorbirdRequestView())
+
+        # This will make HA the only service that gets doorbell events
+        url = hass.config.api.base_url + API_URL + "/" + SENSOR_DOORBELL
+        device.reset_notifications()
+        device.subscribe_notification(SENSOR_DOORBELL, url)
+
+    return True
+
+
+class DoorbirdRequestView(HomeAssistantView):
+    """Provide a page for the device to call."""
+
+    url = API_URL
+    name = API_URL[1:].replace("/", ":")
+    extra_urls = [API_URL + "/{sensor}"]
+
+    @asyncio.coroutine
+    def get(self, request, sensor):
```

Comment:
You can disable the pylint warning `no-self-use`.

Reply:
Example:

https://github.com/home-assistant/home-assistant/blob/fe4e1b71bcd370c53c469df045b1c56c06510123/homeassistant/components/device_tracker/automatic.py#L171